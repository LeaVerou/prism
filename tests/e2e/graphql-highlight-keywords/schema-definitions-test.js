const { expect } = require('../../helper/chai');
const prismFactory = require('../../helper/prism-loader');

const Prism = prismFactory.createInstance(['graphql', 'highlight-keywords']);

describe('E2E test - Prism#highlights schema def\'ns correctly', () => {
	it('schema def ex. 1', () => {
		expect(subject(`
type Human implements Character {
	id: ID!
	name: String!
	friends: [Character]
	appearsIn: [Episode]!
	starships: [Starship]
	totalCredits: Int
}
type Droid implements Character {
	id: ID!
	name: String!
	friends: [Character]
	appearsIn: [Episode]!
	primaryFunction: String
}`)).toMatchSnapshot();
	});

	it('Input Type definition ex. 1', () => {
		expect(subject(`
input OrderInput {
	id: ID!
	customAttributes: [AttributeInput!]
	customer: Customer
	localizationExtensions: [LocalizationExtensionInput!]
	metafields: [MetafieldInput!]
	note: String!
	shippingAddress: MailingAddressInput
	tags: [String!]
}
		`)).toMatchSnapshot();
	});
});

function subject(code, language = 'graphql') {
	return Prism.highlight(code, Prism.languages[language], language);
}
