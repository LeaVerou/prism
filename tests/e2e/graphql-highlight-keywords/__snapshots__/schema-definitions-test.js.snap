// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`E2E test - Prism#highlights schema def'ns correctly Input Type definition ex. 1 1`] = `
<span class="token keyword keyword-input">
  input
</span>
<span class="token class-name atom-input">
  OrderInput
</span>
<span class="token punctuation">
  {
</span>
<span class="token attr-name">
  id
</span>
<span class="token punctuation">
  :
</span>
<span class="token scalar">
  ID
</span>
<span class="token operator">
  !
</span>
<span class="token attr-name">
  customAttributes
</span>
<span class="token punctuation">
  :
</span>
<span class="token punctuation">
  [
</span>
<span class="token class-name atom-input">
  AttributeInput
</span>
<span class="token operator">
  !
</span>
<span class="token punctuation">
  ]
</span>
<span class="token attr-name">
  customer
</span>
<span class="token punctuation">
  :
</span>
<span class="token class-name">
  Customer
</span>
<span class="token attr-name">
  localizationExtensions
</span>
<span class="token punctuation">
  :
</span>
<span class="token punctuation">
  [
</span>
<span class="token class-name atom-input">
  LocalizationExtensionInput
</span>
<span class="token operator">
  !
</span>
<span class="token punctuation">
  ]
</span>
<span class="token attr-name">
  metafields
</span>
<span class="token punctuation">
  :
</span>
<span class="token punctuation">
  [
</span>
<span class="token class-name atom-input">
  MetafieldInput
</span>
<span class="token operator">
  !
</span>
<span class="token punctuation">
  ]
</span>
<span class="token attr-name">
  note
</span>
<span class="token punctuation">
  :
</span>
<span class="token scalar">
  String
</span>
<span class="token operator">
  !
</span>
<span class="token attr-name">
  shippingAddress
</span>
<span class="token punctuation">
  :
</span>
<span class="token class-name atom-input">
  MailingAddressInput
</span>
<span class="token attr-name">
  tags
</span>
<span class="token punctuation">
  :
</span>
<span class="token punctuation">
  [
</span>
<span class="token scalar">
  String
</span>
<span class="token operator">
  !
</span>
<span class="token punctuation">
  ]
</span>
<span class="token punctuation">
  }
</span>
`;

exports[`E2E test - Prism#highlights schema def'ns correctly schema def ex. 1 1`] = `
<span class="token keyword keyword-type">
  type
</span>
<span class="token class-name">
  Human
</span>
<span class="token keyword keyword-implements">
  implements
</span>
<span class="token class-name">
  Character
</span>
<span class="token punctuation">
  {
</span>
<span class="token attr-name">
  id
</span>
<span class="token punctuation">
  :
</span>
<span class="token scalar">
  ID
</span>
<span class="token operator">
  !
</span>
<span class="token attr-name">
  name
</span>
<span class="token punctuation">
  :
</span>
<span class="token scalar">
  String
</span>
<span class="token operator">
  !
</span>
<span class="token attr-name">
  friends
</span>
<span class="token punctuation">
  :
</span>
<span class="token punctuation">
  [
</span>
<span class="token class-name">
  Character
</span>
<span class="token punctuation">
  ]
</span>
<span class="token attr-name">
  appearsIn
</span>
<span class="token punctuation">
  :
</span>
<span class="token punctuation">
  [
</span>
<span class="token class-name">
  Episode
</span>
<span class="token punctuation">
  ]
</span>
<span class="token operator">
  !
</span>
<span class="token attr-name">
  starships
</span>
<span class="token punctuation">
  :
</span>
<span class="token punctuation">
  [
</span>
<span class="token class-name">
  Starship
</span>
<span class="token punctuation">
  ]
</span>
<span class="token attr-name">
  totalCredits
</span>
<span class="token punctuation">
  :
</span>
<span class="token scalar">
  Int
</span>
<span class="token punctuation">
  }
</span>
<span class="token keyword keyword-type">
  type
</span>
<span class="token class-name">
  Droid
</span>
<span class="token keyword keyword-implements">
  implements
</span>
<span class="token class-name">
  Character
</span>
<span class="token punctuation">
  {
</span>
<span class="token attr-name">
  id
</span>
<span class="token punctuation">
  :
</span>
<span class="token scalar">
  ID
</span>
<span class="token operator">
  !
</span>
<span class="token attr-name">
  name
</span>
<span class="token punctuation">
  :
</span>
<span class="token scalar">
  String
</span>
<span class="token operator">
  !
</span>
<span class="token attr-name">
  friends
</span>
<span class="token punctuation">
  :
</span>
<span class="token punctuation">
  [
</span>
<span class="token class-name">
  Character
</span>
<span class="token punctuation">
  ]
</span>
<span class="token attr-name">
  appearsIn
</span>
<span class="token punctuation">
  :
</span>
<span class="token punctuation">
  [
</span>
<span class="token class-name">
  Episode
</span>
<span class="token punctuation">
  ]
</span>
<span class="token operator">
  !
</span>
<span class="token attr-name">
  primaryFunction
</span>
<span class="token punctuation">
  :
</span>
<span class="token scalar">
  String
</span>
<span class="token punctuation">
  }
</span>
`;
